xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/XWikiHibernateStore.java

```diff
@@ -2064,19 +2064,33 @@ private void releaseAllLocksForCurrentUser(final XWikiContext ctx)
 
             // the select clause is compulsory to reach the fullName i.e. the page pointed
             Query query = session
-                .createQuery("select backlink.fullName from XWikiLink as backlink where backlink.id.link = :backlink");
-            query.setString("backlink", this.localEntityReferenceSerializer.serialize(documentReference));
+                .createQuery("select backlink.fullName from XWikiLink as backlink where "
+                    + "backlink.id.link = :backlink");
+
+            WikiReference wikiReference = documentReference.getWikiReference();
+            boolean sameWikiContext = (wikiReference == null || wikiReference.equals(context.getWikiReference()));
+
+            String serializedBacklink;
+
+            // if we are in the same wiki context, we should only get the local reference
+            // but if we are not, then we have to check the full reference, containing the wiki part since
+            // it's how the link are recorded.
+            // This should be changed once the refactoring to support backlinks properly has been done.
+            // See: XWIKI-16192
+            if (sameWikiContext) {
```

Comment:
You are actually reimplementing the "compactwiki" serializer here.

Reply:
ahah, right.